assembly/src/assembler/instruction/u32_ops.rs

```diff
@@ -393,61 +295,51 @@ pub fn u32rotr(
     op_mode: U32OpMode,
     imm: Option<u8>,
 ) -> Result<Option<CodeBlock>, AssemblyError> {
-    match imm {
-        None => match op_mode {
-            U32OpMode::Checked => {
+    match (imm, op_mode) {
+        (Some(imm), U32OpMode::Checked) => {
+            // Assert the top of the stack is a u32 value.
+            // NOTE: We cannot use U32Assert2 since we are potentially pushing a number larger
+            // than u32 for b.
+            // TODO: double check this logic: we shouldn't use U32MUL if both inputs are not
+            // guaranteed to be u32 values
+            validate_param(imm, 0, MAX_U32_ROTATE_VALUE)?;
+            span.push_ops([Pad, U32assert2, Drop, Push(Felt::new(1 << (32 - imm)))]);
+        }
+        (Some(imm), U32OpMode::Unchecked) => {
+            validate_param(imm, 0, MAX_U32_ROTATE_VALUE)?;
+            span.push_op(Push(Felt::new(1 << (32 - imm))));
+        }
+        (None, U32OpMode::Checked) => {
+            #[rustfmt::skip]
+            span.push_ops([
                 // Verify both b and a are u32.
-                span.push_op(U32assert2);
+                U32assert2,
+
                 // Calculate 32 - b and assert that the shift value b <= 31.
-                span.push_op(Push(Felt::new(31)));
-                span.push_op(Dup1);
-                span.push_op(U32sub);
-                span.push_op(Not);
-                span.push_op(Assert);
-                span.push_op(Incr);
-                span.push_op(Dup1);
+                Push(Felt::from(MAX_U32_ROTATE_VALUE)),
+                Dup1,
+                U32sub,
+                Not,
+                Assert,
+                Incr,
+                Dup1,
```

Comment:
minor nit: should these be listed in one row?

Reply:
Fixed